OpenLIFULib/OpenLIFULib/transducer.py

```diff
@@ -194,7 +194,7 @@ def is_matching_transform(self, query_transform_node: vtkMRMLTransformNode) -> b
         query_transform_node.GetMatrixTransformToParent(query_transform_matrix)
 
         is_matching = (
-            slicer.util.arrayFromVTKMatrix(current_transform_matrix) == slicer.util.arrayFromVTKMatrix(query_transform_matrix)
-            ).all()
+            np.isclose(slicer.util.arrayFromVTKMatrix(current_transform_matrix), slicer.util.arrayFromVTKMatrix(query_transform_matrix))
+        ).all()
```

Comment:
Note that I've modified the exact equality check here

Since there could be rounding or precision issues when the floats are read in and out, they could match without being literally equal bit for bit. So here it is better to use `np.isclose` which does a "fuzzy" equality check based on a resonable default for relative and absolute tolerance.